### PR TITLE
fix(date-picker): open and close datepicker after pressing selector button

### LINKED ISSUE #3282: [BUG] - Date Picker not closing after clicking the selector icon again
### NextUI Version

2.4.2

### Describe the bug

when you click the selector icon, the date picker should be open. If you click it again, it should close. 

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

- Have a basic date picker
- click the selector icon, the date picker is open
- click the selector icon again, the date picker is still open while it should be closed

### Expected behavior

It should close after clicking the selector icon again

### Screenshots or Videos

[date-picker-issue-demo.webm](https://github.com/nextui-org/nextui/assets/35857179/ae6737ed-3e4c-4711-8a38-50a9dff43c3e)

### Operating System Version

macOS

### Browser

Chrome

### PR DESCRIPTION
**Pull request by wingkwong**
Closes #3282

## 📝 Description

- fixed the selector button closing issue (applied the current logic from date-range-picker)
- revised date-range-picker trigger ref

## ⛳️ Current behavior (updates)

- press selector button
- datepicker is open
- press selector button again
- datepicker is still open

## 🚀 New behavior

- press selector button
- datepicker is open
- press selector button again
- datepicker is closed

[pr3283-demo.webm](https://github.com/nextui-org/nextui/assets/35857179/c0f8d5fb-88d6-4db5-9844-a16dcb4caad0)

## Summary by CodeRabbit

- **Bug Fixes**
  - Resolved issue where the date picker would close immediately after pressing the selector button.

- **New Features**
  - Introduced support for additional properties (`innerWrapperProps`) in date input and picker components for more flexible customization.

- **Tests**
  - Added a test case to verify the date picker popover behavior when clicking the selector button.

### CODE REVIEW

File: packages/components/date-input/src/use-date-input.ts [hunk 1]
```diff
@@ -246,6 +249,7 @@ export function useDateInput<T extends DateValue>(originalProps: UseDateInputPro
     return {
       ...props,
       "data-slot": "inner-wrapper",
+      ...mergeProps(innerWrapperPropsProp, props),
       className: slots.innerWrapper({
         class: classNames?.innerWrapper,
```
Comment (jrgarciadev) [inline packages/components/date-input/src/use-date-input.ts:254]:
```suggestion
        class: cn(classNames?.innerWrapper, innerWrapperPropsProp?.className, props?.className)
```

Comment (jrgarciadev) [reply packages/components/date-input/src/use-date-input.ts:254]: @wingkwong

**Review by jrgarciadev — APPROVED** [review]

**Comment by jrgarciadev** [comment]
@wingkwong please fix the conflicts

**Comment by wingkwong** [comment]
@jrgarciadev done

**Pull request merged**